v1.5.4: Vento di Freschezza - fresh wind

dag7dev opened the pull request:
Closes #44 - replaces #45 

Sono stato costretto ad effettuare una nuova pr in quanto quella vecchia già presente non poteva essere utilizzata: troppe parti non funzionanti, troppi bug e troppo codice non mantenuto.

Ecco la lista dei cambiamenti:
```
- aggiunti nuovi comandi (quelli richiesti sono stati generalizzati per permettere una più efficiente gestione)
    - Gestione messaggi nei canali:
        - /admin messaggio preview |canale| |Messaggio da inviare in un canale| => Anteprima del messaggio da inviare, per verificare che tutto venga visualizzato correttamente
        - /admin messaggio | canale | |Messaggio da inviare in un canale in particolare|
        - /admin messaggio broadcast |Messaggio da inviare in tutti i canali|

    - Gestione canali:
        - /admin canale mostra
        - /admin canale aggiungi |Channel_name|
        - /admin canale elimina |Channel_name|
```
Originariamente erano stati richiesti:
```
/admin mozitanews <message>
/admin mozitanews preview <message>
/admin canale aggiungi
/admin canale mostra
/admin canale rimuovi
```
- codice riformattato
- refactoring del codice (non è molto, ma è un lavoro onesto, il bot va riscritto da zero ma finchè non viene fatto va bene così) + funzioni di utilità
- migliorata la chiarezza in generale
- adminlist_hub aggiornato: ora contiene gli id degli admin aggiornati
- aggiunte nuove eccezioni in "ECCEZIONI.MD"
- stampa_su_file() ora si chiama "log()" per essere più chiari
- rimossi commenti superflui

**Tutto è stato possibile grazie anche al grandissimo @Sav22999 che ha avuto una pazienza infinita e ha testato il tutto!**

![Non è molto ma è un lavoro onesto](https://www.drcommodore.it/wp-content/uploads/2019/03/photo5810197553365495636.jpg)

GG! Happy Hacktoberfest!

dag7dev committed:
Merge pull request #1 from MozillaItalia/master

merge

dag7dev committed:
v.1.5.0: big update

- added: new commands
    - Gestione messaggi nei canali:
        - /admin messaggio preview |canale| |Messaggio da inviare in un canale| => Anteprima del messaggio da inviare, per verificare che tutto venga visualizzato correttamente
        - /admin messaggio | canale | |Messaggio da inviare in un canale|
        - /admin messaggio broadcast |Messaggio da inviare in tutti i canali|

    - Gestione canali:
        - /admin canale mostra
        - /admin canale aggiungi |Channel_name|
        - /admin canale elimina |Channel_name|

- reformatted code
- refactored code (not so much, but enough) + utility functions
- improved clarity overall
- adminlist_hub updated: now it contains updated ids
- added new exceptions in "ECCEZIONI.MD"
- stampa_su_file() now it is called "log" to be more clear
- removed some useless comments

dag7dev committed:
v1.5.1: fixed misspelling

Mte90 commented on adminlist_hub.json:
```diff
@@ -1 +1 @@
-[240188083,69903837,75870906,295348075,123150516,810740389,161975186]
```
Chi stiamo rimuovendo?

dag7dev replied:
Non lo so siccome non è documentato.
Possiamo sempre però documentare e capire chi sono gli id e perché sono lì.

Mte90 replied:
C'è un modo da un id di ottenere il nickname @Sav22999 ? Così verifichiamo :-)

Sav22999 replied:
@Mte90 Sì, c'è. Puoi usare la funzione `/admin avviso preview` seguito dal codice HTML relativo:
![image](https://user-images.githubusercontent.com/32335378/94945663-4ceecf00-04db-11eb-80a6-a32788463f64.png)

dag7dev replied:
Documentate con la nuova PR e aggiunti nuovamente (l'altro ID era un vecchio id non aggiornato).

dag7dev committed:
v1.5.2 ids doc in comments + clarified smth in admin section

Sav22999 commented:
Ok, faccio alcuni test e vi faccio sapere :smile:

Sav22999 commented:
`/admin messaggio` -> Rinominarlo in `/admin canale`

"Messaggio" potrebbe trarre in inganno e far pensare che sia "Avviso"


Quindi avere:


/admin canale broadcast CIAO CIAO CIAO
/admin canale aggiungi CIAO —> Mette in automatico la "@"
/admin canale elimina CIAO
/admin canale CIAO messaggio Messaggio
/admin canale mostra

Sav22999 commented:
Sto testando il bot e ho trovato alcune migliorie. Ne sto discutendo direttamente con Damiano in chat privata

dag7dev committed:
v1.5.3 - la resa dei conti

- bugfixes
- renamed /admin messaggio in /admin canale: now they belong to a 'single' kind
- fixed: bot used to show improper error message if channel wasnt present
- improved: regardless of user input, everything is converted in lowercase
- improved: added @ in front of a username if needed
- some other things I don't remember...

dag7dev committed:
v1.5.3 - la resa dei conti

- bugfixes
- renamed /admin messaggio in /admin canale: now they belong to a 'single' kind
- fixed: bot used to show improper error message if channel wasnt present
- improved: regardless of user input, everything is converted in lowercase
- improved: added @ in front of a username if needed
- added: config-sample.ini since it didn't exist

dag7dev commented:
Riporto per completezza il testo del commit qui:
```
- bugfixes
- renamed /admin messaggio in /admin canale: now they belong to a 'single' kind
- fixed: bot used to show improper error message if channel wasnt present
- improved: regardless of user input, everything is converted in lowercase
- improved: added @ in front of a username if needed
- added: config-sample.ini since it didn't exist
```

dag7dev committed:
v1.5.4 - fresh wind

- improved: help
